crates/adapters/src/controller/stats.rs

```diff
@@ -565,6 +606,12 @@ pub struct InputEndpointMetrics {
     pub num_parse_errors: AtomicU64,
 
     pub end_of_input: AtomicBool,
+
+    /// The step to which arriving input belongs.
```

Comment:
Done:

```patch
commit 738c4c033558bb07b47ea68e39e15b4570f8ebba
Author: Ben Pfaff <blp@feldera.com>
Date:   Tue Nov 7 14:40:46 2023 -0800

    dbsp_adapters: Better document step numbers.
    
    Signed-off-by: Ben Pfaff <blp@feldera.com>

diff --git a/crates/adapters/src/controller/stats.rs b/crates/adapters/src/controller/stats.rs
index aa2698f0..679eb601 100644
--- a/crates/adapters/src/controller/stats.rs
+++ b/crates/adapters/src/controller/stats.rs
@@ -31,7 +31,7 @@
 //! pending.
 
 use super::{EndpointId, InputEndpointConfig, OutputEndpointConfig, RuntimeConfig};
-use crate::{transport::Step, PipelineState};
+use crate::{transport::{Step, AtomicStep}, PipelineState};
 use anyhow::Error as AnyError;
 use crossbeam::sync::{ShardedLock, ShardedLockReadGuard, Unparker};
 use log::error;
@@ -608,10 +608,10 @@ pub struct InputEndpointMetrics {
     pub end_of_input: AtomicBool,
 
     /// The step to which arriving input belongs.
-    pub step: AtomicU64,
+    pub step: AtomicStep,
 
     /// The first step known not to have committed yet.
-    pub uncommitted: AtomicU64,
+    pub uncommitted: AtomicStep,
 }
 
 /// Input endpoint status information.
diff --git a/crates/adapters/src/transport/mod.rs b/crates/adapters/src/transport/mod.rs
index fe8a2128..a1a61716 100644
--- a/crates/adapters/src/transport/mod.rs
+++ b/crates/adapters/src/transport/mod.rs
@@ -68,6 +68,9 @@ pub mod durable_kafka;
 /// A "durable" data transport divides input into steps numbered sequentially.
 /// The first step is numbered zero.  Each step has durable content; that is, if
 /// it is read again, it will be the same as the first time.
+///
+/// The step number increases by 1 each time the circuit runs; that is, it
+/// tracks the global clock for the outermost circuit.
 pub type Step = u64;
 
 /// Atomic version of [`Step`].
```

> A question of terminology is if something belongs to a "step" or is labeled with a step.

I don't have a position on this grammatical issue.